### PR TITLE
test(upgrade): allow time for helper ns to be deleted in upgrade testing

### PR DESCRIPTION
**Pull request by gunnar-solo**
# Description
Added some "wait until finished deleting" logic around a namespace delete operation.  We hit a bug, recently, where we had a resource conflict from _not_ having said namespace be deleted

# Checklist:
- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -212,6 +195,13 @@ func addSecondGatewayProxySeparateNamespaceTest(crdDir string, testHelper *helpe
 		return exec_utils.RunCommandOutput(testHelper.RootDir, false,
 			"kubectl", "get", "serviceaccount", "-n", externalNamespace)
 	}, "10s", "1s").Should(ContainSubstring("gateway-proxy"))
+
+	// Ensures namespace is cleaned up before continuing
+	runAndCleanCommand("kubectl", "delete", "ns", externalNamespace)
```
Comment (jenshu) [inline test/kube2e/upgrade/upgrade_test.go:200]: if one of the Eventuallys above fails, we won't get here right?

Comment (gunnar-solo) [reply test/kube2e/upgrade/upgrade_test.go:200]: correct

File: test/kube2e/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -140,18 +136,6 @@ var _ = Describe("Kube2e: Upgrade Tests", func() {
 // ===================================
 // Repeated Test Code
 // ===================================
-// Based case test for local runs to help narrow down failures
```
Comment (ianmacclancy) [inline test/kube2e/upgrade/upgrade_test.go:143]: This should remain in as test code - quickly allows you to run the tests without extra values

File: test/kube2e/upgrade/upgrade_test.go [hunk 3]
```diff
@@ -383,7 +359,7 @@ func runAndCleanCommand(name string, arg ...string) []byte {
 		}
 	}
 	Expect(err).To(BeNil())
-	cmd.Process.Kill()
+	cmd.Process.Kill() // This is *almost certainly* the reason a namespace deletion was able to hang without alerting us
```
Comment (ianmacclancy) [inline test/kube2e/upgrade/upgrade_test.go:362]: So skipping the normal wait and moving on before deletion?

**Review by ianmacclancy — CHANGES_REQUESTED** [review]
Please add back in the test methods - other than that looks good

File: test/kube2e/upgrade/upgrade_test.go (hunk 3 above)
Comment (gunnar-solo) [reply test/kube2e/upgrade/upgrade_test.go:362]: that's what I *think* is happening

File: test/kube2e/upgrade/upgrade_test.go (hunk 2 above)
Comment (gunnar-solo) [reply test/kube2e/upgrade/upgrade_test.go:143]: mostly, my IDE was complaining to me.  *I GUESS*

File: test/kube2e/upgrade/upgrade_test.go (hunk 1 above)
Comment (jenshu) [reply test/kube2e/upgrade/upgrade_test.go:216]: hmm, it's probably ok in that case that we don't clean up

**Review by jenshu — APPROVED** [review]

**Pull request merged**